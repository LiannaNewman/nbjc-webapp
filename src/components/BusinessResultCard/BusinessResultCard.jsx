import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import { withStyles } from '@material-ui/core/styles';

import {
  Card, CardContent, Checkbox, Typography,
} from '@material-ui/core';

import { businessProps } from '../../types';

const styles = (theme) => ({
  card: {
    maxWidth: 345,
    maxHeight: 160,
    borderColor: theme.palette.primary.main,
  },
  checked: {
    backgroundColor: `${theme.palette.primary.light}20`,
  },
  title: {
    display: 'inline',
  },
  checkbox: {
    float: 'right',
    padding: 0,
    borderColor: theme.palette.primary.main,
  },
  wrapper: {
    display: 'flex',
    '& h6': {
      fontWeight: 600,
      marginRight: 15,
    },
  },
});

const BusinessResultCard = ({
  business, classes, checked, onCheck, overrideClasses, hideCheck,
}) => (
  <div className={cx(classes.resultCard, overrideClasses.resultCard)}>
    <Card className={cx(classes.card, { [classes.checked]: checked })} variant="outlined">
      <CardContent>
        <div>
          <Typography variant="subtitle1" className={classes.title}>{business.name}</Typography>
          {!hideCheck && <Checkbox color="primary" className={classes.checkbox} checked={checked} onChange={onCheck} /> }
        </div>
        <Typography variant="caption">{business.category}</Typography>
        <div className={classes.wrapper}>
          <Typography variant="subtitle2">Address:</Typography>
          <Typography variant="body2">{business.address}</Typography>
        </div>
        <div className={classes.wrapper}>
          <Typography variant="subtitle2">Phone number:</Typography>
          <Typography variant="body2">{business.phoneNumber}</Typography>
        </div>
      </CardContent>
    </Card>
  </div>
);

BusinessResultCard.propTypes = {
  business: PropTypes.shape(businessProps).isRequired,
  checked: PropTypes.bool.isRequired,
  classes: PropTypes.shape({}).isRequired,
  overrideClasses: PropTypes.shape({}),
  hideCheck: PropTypes.bool,
};

BusinessResultCard.defaultProps = {
  overrideClasses: {},
  hideCheck: false,
};

export default withStyles(styles)(BusinessResultCard);
