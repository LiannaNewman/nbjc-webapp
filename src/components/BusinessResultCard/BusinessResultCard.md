## BusinessResultCard

```jsx
const [checked, setChecked] = React.useState(false);
const business =  {
  id: '123',
  name: 'La colombe coffee roasters',
  category: 'Coffee',
  address: '924 Blagden Alley Way Washington, D.C. 20001',
  distance: '0.03mi',
  phoneNumber: '123-456-7890',
};
<BusinessResultCard business={business} checked={checked} onCheck={() => setChecked(!checked)} />
```