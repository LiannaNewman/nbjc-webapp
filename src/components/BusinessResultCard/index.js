import BusinessResultCard from './BusinessResultCard';

export default BusinessResultCard;
