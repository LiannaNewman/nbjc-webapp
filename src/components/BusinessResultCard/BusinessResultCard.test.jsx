import React from 'react';
import { render } from '@testing-library/react';

import BusinessResultCard from './BusinessResultCard';

describe('BusinessResultCard', () => {
  test('renders', () => {
    const onCheck=jest.fn();
    const business = {
      id: '123',
      name: 'La colombe coffee roasters',
      category: 'Coffee',
      address: '924 Blagden Alley Way Washington, D.C. 20001',
      distance: '0.03mi',
      phoneNumber: '123-456-7890',
    };
    const { asFragment } = render(
      <BusinessResultCard
        business={business}
        onCheck={onCheck}
        checked={false}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
