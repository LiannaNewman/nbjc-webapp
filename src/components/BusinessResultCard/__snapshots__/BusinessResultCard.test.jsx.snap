// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BusinessResultCard renders 1`] = `
<DocumentFragment>
  <div
    class=""
    data-testid="businessresultcard-123"
  >
    <div
      class="MuiPaper-root MuiCard-root BusinessResultCard-card-1 MuiPaper-outlined MuiPaper-rounded"
    >
      <div
        class="MuiCardContent-root"
      >
        <div>
          <h6
            class="MuiTypography-root BusinessResultCard-title-3 MuiTypography-subtitle1"
          >
            La colombe coffee roasters
          </h6>
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-6 MuiCheckbox-root MuiCheckbox-colorPrimary BusinessResultCard-checkbox-4 MuiIconButton-colorPrimary"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                class="PrivateSwitchBase-input-9"
                data-indeterminate="false"
                type="checkbox"
                value=""
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
        </div>
        <span
          class="MuiTypography-root MuiTypography-caption"
        >
          Coffee
        </span>
        <div
          class="BusinessResultCard-wrapper-5"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle2"
          >
            Address:
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body2"
          >
            924 Blagden Alley Way Washington, D.C. 20001
          </p>
        </div>
        <div
          class="BusinessResultCard-wrapper-5"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle2"
          >
            Phone number:
          </h6>
          <p
            class="MuiTypography-root MuiTypography-body2"
          >
            123-456-7890
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
